### PR TITLE
Add tag adding and editing functionality

### LINKED ISSUE #62: As a user, I can add a tag to a dengue case with its dengue variant

### LINKED ISSUE #63: As a user, I can edit the dengue variant tag of a dengue case in the dengue case list

### PR DESCRIPTION
**Pull request by valerietanhx**
Users are unable to tag a dengue case with the associated dengue variant.

Adding dengue variant tags will allow for more granular analysis of the commonness of different dengue strains in Singapore, and enable quick action should a particular strain that residents are less immune to become unusually common.

Let’s add the ability to add and edit tags that will indicate the dengue variant associated with each dengue case.

Using an enum will allow us to easily validate that only actual dengue variants are used as tags compared to using regex or a list of accepted input.

Closes #62, closes #63

### CODE REVIEW

File: src/test/java/seedu/dengue/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -45,14 +45,14 @@ public class CommandTestUtil {
...
-    public static final String TAG_DESC_HUSBAND = " " + PREFIX_TAG + VALID_TAG_HUSBAND;
+    public static final String TAG_DESC_DENV2 = " " + PREFIX_TAG + VALID_TAG_DENV2;
+    public static final String TAG_DESC_DENV1 = " " + PREFIX_TAG + VALID_TAG_DENV1;
 
     public static final String INVALID_NAME_DESC = " " + PREFIX_NAME + "James&"; // '&' not allowed in names
     public static final String INVALID_POSTAL_DESC = " " + PREFIX_POSTAL + "911a"; // 'a' not allowed in postals
     public static final String INVALID_DATE_DESC = " " + PREFIX_DATE + "bob!yahoo"; // missing '@' symbol
     public static final String INVALID_AGE_DESC = " " + PREFIX_AGE; // empty string not allowed for age
-    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "hubby*"; // '*' not allowed in tags
+    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "sever"; // "severe" not allowed in tags
```
Comment (axmszr) [inline src/test/java/seedu/dengue/logic/commands/CommandTestUtil.java:55]: Should the comment be changed to more accurately reflect why this tag is invalid? e.g. `// "sever" is not an accepted tag`

File: src/test/data/JsonSerializableDengueHotspotTrackerTest/invalidPersonDengueHotspotTracker.json [hunk 2]
```diff
@@ -3,6 +3,7 @@
     "name": "Hans Muster",
     "postal": "948424a",
     "date": "2022-11-02",
-    "age": "21"
+    "age": "21",
+    "tag": "DENV1"
```
Comment (axmszr) [inline src/test/data/JsonSerializableDengueHotspotTrackerTest/invalidPersonDengueHotspotTracker.json:7]: Is the format invalid on purpose?

```suggestion
    "tagged": [ "DENV1" ]
```

File: src/test/data/JsonDengueHotspotTrackerStorageTest/invalidPersonDengueHotspotTracker.json [hunk 3]
```diff
@@ -3,6 +3,7 @@
     "name": "Person with invalid name field: Ha!ns Mu@ster",
     "postal": "948242",
     "date": "2023-03-05",
-    "age": "23"
+    "age": "23",
+    "tag": "DENV1"
```
Comment (axmszr) [inline src/test/data/JsonDengueHotspotTrackerStorageTest/invalidPersonDengueHotspotTracker.json:7]: Is the tag format invalid on purpose?

```suggestion
    "tagged": [ "DENV1" ]
```

**Review by axmszr — CHANGES_REQUESTED** [review]
Mostly just a few nits about `"tag"` vs. `"tagged"` in the JSONs. Else LGTM!

File: src/test/java/seedu/dengue/logic/commands/CommandTestUtil.java (hunk 1 above)
Comment (valerietanhx) [reply src/test/java/seedu/dengue/logic/commands/CommandTestUtil.java:55]: Good call!

```suggestion
    public static final String INVALID_TAG_DESC = " " + PREFIX_TAG + "severe"; // "severe" is not a valid tag
```

File: src/test/data/JsonDengueHotspotTrackerStorageTest/invalidPersonDengueHotspotTracker.json (hunk 3 above)
Comment (valerietanhx) [reply src/test/data/JsonDengueHotspotTrackerStorageTest/invalidPersonDengueHotspotTracker.json:7]: Oops, looks like I missed the brackets around the tags!! Thanks for catching that

File: src/test/data/JsonDengueHotspotTrackerStorageTest/invalidAndValidPersonDengueHotspotTracker.json [hunk 4]
```diff
@@ -3,11 +3,13 @@
     "name": "Valid Person",
     "postal": "948242",
     "date": "2023-03-05",
-    "age": "22"
+    "age": "22",
+    "tag": "DENV1"
```
[the remark this answers is not in the document]
Comment (valerietanhx) [reply src/test/data/JsonDengueHotspotTrackerStorageTest/invalidAndValidPersonDengueHotspotTracker.json:7]: Yup!! Thanks for spotting :")

File: src/test/data/JsonDengueHotspotTrackerStorageTest/invalidAndValidPersonDengueHotspotTracker.json [hunk 5]
```diff
@@ -3,11 +3,13 @@
...
   }, {
     "name": "Person With Invalid Postal Field",
     "postal": "948asdf2424",
     "date": "2023-03-05",
-    "age": "22"
+    "age": "22",
+    "tag": "DENV1"
```
[the remark this answers is not in the document]
Comment (valerietanhx) [reply src/test/data/JsonDengueHotspotTrackerStorageTest/invalidAndValidPersonDengueHotspotTracker.json:13]:
```suggestion
    "tagged": [ "DENV1" ]
```

**Review by axmszr — APPROVED** [review]

**Review by axmszr — APPROVED** [review]
LGTM!

**Pull request merged**